db/batchlog_manager.cc

```diff
@@ -242,7 +242,8 @@ future<> db::batchlog_manager::replay_all_failed_batches(post_replay_cleanup cle
                 // send to partially or wholly fail in actually sending stuff. Since we don't
                 // have hints (yet), send with CL=ALL, and hope we can re-do this soon.
                 // See below, we use retry on write failure.
-                return _qp.proxy().mutate(mutations, db::consistency_level::ALL, db::no_timeout, nullptr, empty_service_permit(), db::allow_per_partition_rate_limit::no);
+                auto timeout = db::timeout_clock::now() + _write_request_timeout;
+                return _qp.proxy().mutate(mutations, db::consistency_level::ALL, timeout, nullptr, empty_service_permit(), db::allow_per_partition_rate_limit::no);
```

Comment:
The problem with using a timeout here is that it's not clear what the timeout should be. Imagine that you configured the node with a small write timeout and writing the batch initially fails due to write timeout - because it was just big and applying it took a lot of time. IIUC, the retry logic here will try to use the same value for the timeout here, so in this particular case batchlog replay will get stuck.

I'm not really sure what should be a good value for the timeout here.

We had the same issues with respect to hint replay in the past and IIRC the solution to the problem was to abort in-flight hint writes when shutting down. How difficult would it be to implement this for batch writes?

Reply:
@denesb may have an idea.